agnhost postsubmit build failing

<!-- Please only use this template for submitting reports about continuously failing tests or jobs in Kubernetes CI -->

#### Which jobs are failing:

post-kubernetes-push-e2e-agnhost-test-images

#### Since when has it been failing: 4/30

#### Testgrid link: https://testgrid.k8s.io/sig-testing-images#post-kubernetes-push-e2e-agnhost-test-images

#### Reason for failure:

Image build unhappy on s390x arch: 

```
Building image for agnhost OS/ARCH: linux/s390x...

[...]

#7 [ 3/10] RUN apk --update add bind-tools curl netcat-openbsd iproute2 ipe...
#7 0.165 fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/s390x/APKINDEX.tar.gz
#7 0.725 fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/s390x/APKINDEX.tar.gz
#7 1.719 (1/35) Installing ncurses-terminfo-base (6.2_p20200523-r0)
#7 1.758 (2/35) Installing ncurses-libs (6.2_p20200523-r0)
#7 1.811 (3/35) Installing readline (8.0.4-r0)
#7 1.851 (4/35) Installing bash (5.0.17-r0)
#7 1.939 Executing bash-5.0.17-r0.post-install
#7 1.945 ERROR: bash-5.0.17-r0.post-install: script exited with error 1
#7 1.945 (5/35) Installing fstrm (0.6.0-r1)
#7 1.987 (6/35) Installing libgcc (9.3.0-r2)
#7 2.013 (7/35) Installing krb5-conf (1.0-r2)
#7 2.053 (8/35) Installing libcom_err (1.45.6-r0)
#7 2.093 (9/35) Installing keyutils-libs (1.6.1-r1)
#7 2.133 (10/35) Installing libverto (0.3.1-r1)
#7 2.174 (11/35) Installing krb5-libs (1.18.3-r0)
#7 2.334 (12/35) Installing json-c (0.14-r1)
#7 2.376 (13/35) Installing libstdc++ (9.3.0-r2)
#7 2.493 (14/35) Installing libprotobuf (3.12.2-r0)
#7 2.667 (15/35) Installing libprotoc (3.12.2-r0)
#7 2.807 (16/35) Installing protobuf-c (1.3.3-r1)
#7 2.854 (17/35) Installing libuv (1.38.1-r0)
#7 2.906 (18/35) Installing xz-libs (5.2.5-r0)
#7 2.954 (19/35) Installing libxml2 (2.9.10-r5)
#7 3.071 (20/35) Installing bind-libs (9.16.15-r0)
#7 3.287 (21/35) Installing bind-tools (9.16.15-r0)
#7 3.363 (22/35) Installing ca-certificates (20191127-r4)
#7 3.426 (23/35) Installing nghttp2-libs (1.41.0-r0)
#7 3.456 (24/35) Installing libcurl (7.76.1-r0)
#7 3.510 (25/35) Installing curl (7.76.1-r0)
#7 3.561 (26/35) Installing iperf (2.0.13-r2)
#7 3.607 (27/35) Installing libbz2 (1.0.8-r1)
#7 3.649 (28/35) Installing fts (1.2.7-r1)
#7 3.691 (29/35) Installing libelf (0.179-r0)
#7 3.774 (30/35) Installing libmnl (1.0.4-r0)
#7 3.814 (31/35) Installing libnftnl-libs (1.1.6-r0)
#7 3.865 (32/35) Installing iptables (1.8.4-r2)
#7 4.387 (33/35) Installing iproute2 (5.6.0-r0)
#7 4.535 Executing iproute2-5.6.0-r0.post-install
#7 4.540 ERROR: iproute2-5.6.0-r0.post-install: script exited with error 1
#7 4.540 (34/35) Installing libbsd (0.10.0-r0)
#7 4.584 (35/35) Installing netcat-openbsd (1.130-r1)
#7 4.627 Executing busybox-1.31.1-r20.trigger
#7 4.632 ERROR: busybox-1.31.1-r20.trigger: script exited with error 1
#7 4.632 Executing ca-certificates-20191127-r4.trigger
#7 4.686 2 errors; 29 MiB in 49 packages
#7 ERROR: executor failed running [/dev/.buildkit_qemu_emulator /bin/sh -c apk --update add bind-tools curl netcat-openbsd iproute2 iperf bash && rm -rf /var/cache/apk/*   && ln -s /usr/bin/iperf /usr/local/bin/iperf   && ls -altrh /usr/local/bin/iperf]: exit code: 2
------
 > [ 3/10] RUN apk --update add bind-tools curl netcat-openbsd iproute2 iperf bash && rm -rf /var/cache/apk/*   && ln -s /usr/bin/iperf /usr/local/bin/iperf   && ls -altrh /usr/local/bin/iperf:
------
failed to solve: rpc error: code = Unknown desc = executor failed running [/dev/.buildkit_qemu_emulator /bin/sh -c apk --update add bind-tools curl netcat-openbsd iproute2 iperf bash && rm -rf /var/cache/apk/*   && ln -s /usr/bin/iperf /usr/local/bin/iperf   && ls -altrh /usr/local/bin/iperf]: exit code: 2
make: *** [Makefile:43: all-build-and-push] Error 1
ERROR
ERROR: build step 0 "gcr.io/k8s-testimages/gcb-docker-gcloud:v20201130-750d12f" failed: step exited with non-zero status: 2
ERROR: (gcloud.builds.submit) build 32ba795f-78ae-4086-bc26-813044ebaa8c completed with status "FAILURE"
--------------------------------------------------------------------------------


2021/04/30 16:58:31 Failed to run some build jobs: [error running [gcloud builds submit --verbosity info --config /home/prow/go/src/github.com/kubernetes/kubernetes/test/images/cloudbuild.yaml --substitutions _PULL_BASE_REF=master,_WHAT=agnhost,_GIT_TAG=v20210430-v1.22.0-alpha.1-58-gf039f94a6fe --project k8s-staging-e2e-test-images --gcs-log-dir gs://k8s-staging-e2e-test-images-gcb/logs --gcs-source-staging-dir gs://k8s-staging-e2e-test-images-gcb/source gs://k8s-staging-e2e-test-images-gcb/source/2b005587-1c8a-4141-9bbb-f69391463f5c.tgz]: exit status 1]
```

#### Anything else we need to know:

Need this fixed to finish image version dance to fix conformance tests for 1.21.1: https://github.com/kubernetes/kubernetes/pull/101589

--- a/test/images/agnhost/Dockerfile
+++ b/test/images/agnhost/Dockerfile
@@ -27,7 +27,8 @@ CROSS_BUILD_COPY qemu-QEMUARCH-static /usr/bin/
 # - iproute2: includes ss used in NodePort tests
 # from iperf image
 # install necessary packages: iperf, bash
-RUN apk --update add bind-tools curl netcat-openbsd iproute2 iperf bash && rm -rf /var/cache/apk/* \
+RUN retry () { i=0; while [ $i -lt 9 ]; do "$@" && return || sleep 30; i="${i+1}"; done; "$@"; } \
+  && retry apk --update add bind-tools curl netcat-openbsd iproute2 iperf bash && rm -rf /var/cache/apk/* \
   && ln -s /usr/bin/iperf /usr/local/bin/iperf \
   && ls -altrh /usr/local/bin/iperf
 

